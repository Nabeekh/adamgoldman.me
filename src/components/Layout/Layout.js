// @flow

import React from 'react'
import withStyles from 'isomorphic-style-loader/lib/withStyles'
import axios from 'axios'

import {
  TYPEFORM_ID_SAVORING_INTRO,
  // TYPEFORM_ID_SAVORING_PEACEFUL_ENDING,
  // TYPEFORM_ID_SAVORING_REENGAGING_THE_FUTURE,
  // TYPEFORM_ID_SAVORING_RELATIONSHIP_CONSOLIDATION,
  // TYPEFORM_ID_SAVORING_REUNION,
  // TYPEFORM_ID_SAVORING_SAVORING_THE_FUTURE,
  // TYPEFORM_ID_SAVORING_SPECIAL_DAYS,
} from '../../constants'
import history from '../../history'
import Footer from '../Footer'
import MainNav from '../MainNav'
import MessengerFixed from '../MessengerFixed'

import s from './Layout.css'

type Props = {
  children: any,
  path: string,
}

class Layout extends React.Component {
  state = {
    user: {},
  }

  componentDidMount() {
    const user = this.syncUserFromLS()
    this.gateKeeper(user)
  }

  props: Props

  render() {
    const { children, path } = this.props
    const { user } = this.state
    return (
      <div>
        <MainNav path={path} user={user} onLogin={this.login} onLogout={this.logout} />
        <div>
          {React.cloneElement(children, {
            user,
            onLogin: this.login,
            onSubmitIntro: this.submitIntro,
            onSubmitModule: this.submitModule,
          })}
        </div>
        <div className="container">
          <Footer />
        </div>
        <MessengerFixed path={path} />
      </div>
    )
  }

  syncUserFromLS() {
    let user = localStorage.getItem('user')
    if (user) {
      user = JSON.parse(user)
      global.console.log('user logged in', user)
      this.setState({ user })
    }
    return user || {}
  }

  login = (user) => {
    global.console.log('user login', user)
    this.updateUser(user)
  }

  submitIntro = ({ gender, childName, genderParent }) => {
    const user = Object.assign(
      {},
      this.state.user,
      { gender, childName, genderParent },
      { form: [TYPEFORM_ID_SAVORING_INTRO] },
    )
    this.updateUser(user)
    axios.put(`/api/users/${user._id}`, user)
      .then(history.push('/savoring-your-child/test1'))
      .catch(err => global.console.error(err))
  }

  submitModule = async (formId) => {
    global.console.log('submitted form', formId)
    const user = Object.assign({}, this.state.user)
    user.form.push(formId)
    this.updateUser(user)
    axios.put(`/api/users/form/${user._id}`, formId)
      .catch(err => global.console.error(err))
    try {
      const formResponses = await axios.get(`api/typeform/${formId}`)
      const formData = formResponses.data.responses.filter(res => res.hidden.user_id === user._id)[0].answers // eslint-disable-line max-len
      global.console.log('formData', formData)
    } catch (err) {
      global.alert('there was an error, please contact me, sorry for this!')
      global.console.error('err', err)
    }
    // if (objections) {
    //   openMessengerBotSavoringConcern(formId)
    //   return
    // }
    history.push('/savoring-your-child/modules')
  }

  logout = () => {
    global.console.log('user logout')
    localStorage.removeItem('user')
    this.setState({ user: {} })
  }

  updateUser(user) {
    this.setState({ user })
    localStorage.setItem('user', JSON.stringify(user))
  }

  gateKeeper(user) {
    if (!this.props.path.match(/peaceful-ending|reengaging-the-future|relationship-consolidation|reunion|savoring-the-future|special-days|test1|test2/)) {
      return
    }
    if (user.form && user.form.includes(TYPEFORM_ID_SAVORING_INTRO)) {
      return
    }
    history.push('/savoring-your-child')
  }
}

export default withStyles(s)(Layout)

/* eslint-disable max-len */
// function openMessengerBotSavoringConcern(formId) {
//   let formName
//   if (formId === TYPEFORM_ID_SAVORING_PEACEFUL_ENDING) { formName = 'peaceful-ending' }
//   if (formId === TYPEFORM_ID_SAVORING_REENGAGING_THE_FUTURE) { formName = 'reengaging-the-future' }
//   if (formId === TYPEFORM_ID_SAVORING_RELATIONSHIP_CONSOLIDATION) { formName = 'relationship-consolidation' }
//   if (formId === TYPEFORM_ID_SAVORING_REUNION) { formName = 'reunion' }
//   if (formId === TYPEFORM_ID_SAVORING_SAVORING_THE_FUTURE) { formName = 'savoring-the-future' }
//   if (formId === TYPEFORM_ID_SAVORING_SPECIAL_DAYS) { formName = 'special-days' }
//   window.open(`https://m.me/adamgoldman.me?ref=${formName}-concern`, '_newtab')
// }
/* eslint-enable max-len */
