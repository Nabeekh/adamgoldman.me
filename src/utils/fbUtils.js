import axios from 'axios'

import { DOMAIN } from '../constants'

export const fbAuth = (cb) => {
  global.FB.login(res => fbAuth2(res, cb), { scope: 'email,public_profile' })
}

export const getFbShareUrl = path => `https://www.facebook.com/sharer/sharer.php?u=${DOMAIN}${path}`

function fbAuth2(response, cb) {
  if (response.status !== 'connected') {
    global.alert('please authorize login to continue')
    return
  }
  const { accessToken, userID } = response.authResponse
  global.FB.api('/me?fields=id,name,picture',
    ({ name, picture }) => {
      userServerLogin({
        fbUserId: userID, fbClientAccessToken: accessToken, name, picture,
      }, cb)
    },
  )
}

function userServerLogin(user, cb) {
  if (user.picture && user.picture.data && user.picture.data.url) {
    user.fbPictureUrl = user.picture.data.url
  }
  axios.post('/api/users/fbAuth', user)
    .then((serverRes) => { cb(serverRes.data) })
    .catch((err) => {
      global.console.error(err)
      global.alert('there was an error, please contact me')
    })
}
