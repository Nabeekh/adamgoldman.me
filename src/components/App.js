/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react'
import PropTypes from 'prop-types'
import axios from 'axios'

import { isProd, isMobile } from '../utils'
import { FB_APP_ID } from '../constants'

axios.defaults.baseURL = isProd ? 'http://www.adamgoldman.me' : 'http://localhost:3000'

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/kriasoft/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
}

/**
 * The top-level React component setting context (global) variables
 * that can be accessed from all the child components.
 *
 * https://facebook.github.io/react/docs/context.html
 *
 * Usage example:
 *
 *   const context = {
 *     history: createBrowserHistory(),
 *     store: createStore(),
 *   };
 *
 *   ReactDOM.render(
 *     <App context={context}>
 *       <div>
 *         <LandingPage />
 *       </div>
 *     </App>,
 *     container,
 *   );
 */

/* eslint-disable */
const initFbSdk = () => {
  window.fbAsyncInit = function() {
    FB.init({
      appId            : FB_APP_ID,
      autoLogAppEvents : true,
      xfbml            : true,
      version          : 'v2.11'
    });
  };

  (function(d, s, id){
     var js, fjs = d.getElementsByTagName(s)[0];
     if (d.getElementById(id)) {return;}
     js = d.createElement(s); js.id = id;
     js.src = "https://connect.facebook.net/en_US/sdk.js";
     fjs.parentNode.insertBefore(js, fjs);
   }(document, 'script', 'facebook-jssdk'));
};
/* eslint-enable */

const setAdminPass = () => {
  axios.defaults.headers.common['admin-pass'] = localStorage.getItem('adminPass')
}

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context
  }

  componentDidMount() {
    initFbSdk()
    setAdminPass()
    axios.defaults.headers.common['is-mobile'] = isMobile()
  }

  render() {
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    return React.Children.only(this.props.children)
  }
}

export default App
