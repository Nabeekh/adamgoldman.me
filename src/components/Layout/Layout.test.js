import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'
import { initFbSdk } from '../../utils/fbUtils'

import Layout from './Layout'


jest.mock('../../utils/fbUtils')

describe('Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {} }}>
          <Layout path="/test">
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
    expect(initFbSdk).toHaveBeenCalled()
  })
})
