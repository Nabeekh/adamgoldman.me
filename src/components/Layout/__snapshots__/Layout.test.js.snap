// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-lg fixed-top main-nav navbar-light"
  >
    <div
      className="container"
    >
      <div
        className="clearfix d-sm-block d-md-none"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          onClick={[Function]}
          role="img"
          style={
            Object {
              "marginRight": 10,
            }
          }
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="mobileMenu"
        >
          <nav>
            <h3
              className="mobileMenuHeadline"
              onClick={[Function]}
            >
              Adam Goldman
            </h3>
            <ul
              className="navbar-nav"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/transcripts"
                  onClick={[Function]}
                >
                  Transcripts
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/tools"
                  onClick={[Function]}
                >
                  Tutorials
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/about"
                  onClick={[Function]}
                >
                  About
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="https://www.adamgoldman.me/contact"
                  rel="nofollow noreferrer noopener"
                  target="_blank"
                >
                  Contact
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/i-dont-charge-i-accept"
                  onClick={[Function]}
                >
                  <span>
                    Pricing 
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-heart fa-w-18 "
                      data-icon="heart"
                      data-prefix="fas"
                      role="img"
                      style={
                        Object {
                          "color": "red",
                          "marginLeft": 5,
                        }
                      }
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M414.9 24C361.8 24 312 65.7 288 89.3 264 65.7 214.2 24 161.1 24 70.3 24 16 76.9 16 165.5c0 72.6 66.8 133.3 69.2 135.4l187 180.8c8.8 8.5 22.8 8.5 31.6 0l186.7-180.2c2.7-2.7 69.5-63.5 69.5-136C560 76.9 505.7 24 414.9 24z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                </a>
              </li>
              <li
                className="nav-item"
                style={
                  Object {
                    "marginTop": 20,
                  }
                }
              >
                <a
                  className="nav-link btn btn-primary btn-block"
                  href="https://www.adamgoldman.me/book-session"
                  rel="nofollow noreferrer noopener"
                  target="_blank"
                >
                  Book a session
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <a
        className="navbar-brand mr-auto"
        href="/"
        onClick={[Function]}
      >
        Adam Goldman
      </a>
      <div
        className="collapse navbar-collapse"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/transcripts"
              onClick={[Function]}
            >
              Transcripts
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/tools"
              onClick={[Function]}
            >
              Tutorials
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www.adamgoldman.me/contact"
              rel="nofollow noreferrer noopener"
              target="_blank"
            >
              Contact
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/i-dont-charge-i-accept"
              onClick={[Function]}
            >
              <span>
                Pricing 
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-18 "
                  data-icon="heart"
                  data-prefix="fas"
                  role="img"
                  style={
                    Object {
                      "color": "red",
                      "marginLeft": 5,
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M414.9 24C361.8 24 312 65.7 288 89.3 264 65.7 214.2 24 161.1 24 70.3 24 16 76.9 16 165.5c0 72.6 66.8 133.3 69.2 135.4l187 180.8c8.8 8.5 22.8 8.5 31.6 0l186.7-180.2c2.7-2.7 69.5-63.5 69.5-136C560 76.9 505.7 24 414.9 24z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </span>
            </a>
          </li>
          <li
            className="nav-item"
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            <a
              className="nav-link btn btn-primary btn-sm"
              href="https://www.adamgoldman.me/book-session"
              rel="nofollow noreferrer noopener"
              target="_blank"
            >
              Book a session
            </a>
          </li>
        </ul>
      </div>
      <a
        className="nav-link btn btn-primary btn-sm d-sm-block d-md-none"
        href="https://www.adamgoldman.me/book-session"
        rel="nofollow noreferrer noopener"
        target="_blank"
      >
        Book a session
      </a>
    </div>
  </nav>
  <div>
    <div
      className="child"
      onLogin={[Function]}
      onSubmitIntro={undefined}
      onSubmitModule={[Function]}
      user={Object {}}
    />
  </div>
  <div
    className="container"
  >
    <div
      className="footer"
    >
      <p
        className="pull-left"
      >
        Copyright Â© 2017 
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Adam Goldman
        </a>
      </p>
      <p
        className="pull-right"
      >
        <a
          className="link"
          href="/legal-stuff"
          onClick={[Function]}
        >
          legal stuff
        </a>
        <a
          className="link"
          href="/privacy"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          Privacy
        </a>
      </p>
      <div
        className="clearfix"
      />
    </div>
  </div>
  <a
    className="container"
    href="https://www.adamgoldman.me/contact"
    rel="nofollow noreferrer noopener"
    target="_blank"
  >
    <img
      alt="Messenger Link"
      src="https://cdn.supple.com.au/wp-content/themes/supple/img/msg.png"
    />
  </a>
</div>
`;
